File: baekjoon/Test_10828.py
import sys

n = int(sys.stdin.readline())
stack = []

for _ in range(n):
    word = sys.stdin.readline().split()
    if word[0] == 'push':
        stack.append(word[1])
    elif word[0] == 'pop':
        if stack:
            pop = stack.pop()
            print(pop)
        else:
            print(-1)
    elif word[0] == 'size':
        print(len(stack))
    elif word[0] == 'empty':
        if stack:
            print(0)
        else:
            print(1)
    elif word[0] == 'top':
        if stack:
            print(stack[-1])
        else:
            print(-1)
    else:
        pass

File: baekjoon/Test_10828.java
public class Test_10828 {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        Stack<String> stack = new Stack<>();

        for (int i = 0; i < n; i++) {
            String command = br.readLine();
            StringTokenizer st = new StringTokenizer(command);
            String word = st.nextToken();

            if (word.equals("push")) {
                String value = st.nextToken();
                stack.push(value);
            } else if (word.equals("pop")) {
                if (stack.isEmpty()) {
                    System.out.println(-1);
                } else {
                    System.out.println(stack.pop());
                }
            } else if (word.equals("size")) {
                System.out.println(stack.size());
            } else if (word.equals("empty")) {
                if (stack.isEmpty()) {
                    System.out.println(1);
                } else {
                    System.out.println(0);
                }
            } else if (word.equals("top")) {
                if (stack.isEmpty()) {
                    System.out.println(-1);
                } else {
                    System.out.println(stack.peek());
                }
            }
        }
    }
}